pkgs/development/go-modules/generic/old.nix

```diff
@@ -1,229 +0,0 @@
-{ go, cacert, git, lib, removeReferencesTo, stdenv }:
-
-{ name ? "${args'.pname}-${args'.version}"
-, src
-, buildInputs ? []
-, nativeBuildInputs ? []
-, passthru ? {}
-, patches ? []
-
-# A function to override the go-modules derivation
-, overrideModAttrs ? (_oldAttrs : {})
-
-# path to go.mod and go.sum directory
-, modRoot ? "./"
-
-# modSha256 is the sha256 of the vendored dependencies
-#
-# CAUTION: if `null` is used as a value, the derivation won't be a
-# fixed-output derivation but disable the build sandbox instead. Don't use
-# this in nixpkgs as Hydra won't build those packages.
-, modSha256
-
-# We want parallel builds by default
-, enableParallelBuilding ? true
-
-# Disabled flag
-, disabled ? false
-
-# Do not enable this without good reason
-# IE: programs coupled with the compiler
-, allowGoReference ? false
-
-, meta ? {}
-
-, ... }@args':
-
-with builtins;
-
-let
-  args = removeAttrs args' [ "overrideModAttrs" "modSha256" "disabled" ];
-
-  removeReferences = [ ] ++ lib.optional (!allowGoReference) go;
-
-  removeExpr = refs: ''remove-references-to ${lib.concatMapStrings (ref: " -t ${ref}") refs}'';
-
-  go-modules = go.stdenv.mkDerivation (let modArgs = {
-    name = "${name}-go-modules";
-
-    nativeBuildInputs = [ go git cacert ];
-
-    inherit (args) src;
-    inherit (go) GOOS GOARCH;
-
-    patches = args.patches or [];
-
-    GO111MODULE = "on";
-
-    impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
-      "GIT_PROXY_COMMAND" "SOCKS_SERVER"
-    ];
-
-    configurePhase = args.modConfigurePhase or ''
-      runHook preConfigure
-
-      export GOCACHE=$TMPDIR/go-cache
-      export GOPATH="$TMPDIR/go"
-      mkdir -p "''${GOPATH}/pkg/mod/cache/download"
-      cd "${modRoot}"
-      runHook postConfigure
-    '';
-
-    buildPhase = args.modBuildPhase or ''
-      runHook preBuild
-
-      go mod download
```

Comment:
Currently `go mod vendor` prunes non-package directories https://github.com/golang/go/issues/26366, which caused various regressions  https://github.com/NixOS/nixpkgs/issues/89310 https://github.com/NixOS/nixpkgs/issues/89128 , and `vend` might not solve some cases https://github.com/NixOS/nixpkgs/pull/89453

So, I was thinking that we could use `go mod download` to generate the vendor directory, which seem to be the recommended way to vendorize https://github.com/go-modules-by-example/index/blob/master/012_modvendor/README.md. What do you think?

Reply:
Maybe I've misunderstood what you've proposing but I don't think the generated output of `download` is compatible with `-mod=vendor` so we'd need to work around it somehow which is basically what the original approach was doing?

Reply:
> I don't think the generated output of download is compatible with -mod=vendor

oh, you are right! they aren't compatible. I thought `go mod download` could create a `vendor/`, sorry about the noise